### PR TITLE
[FrameworkBundle] remove translation data collector when not usable

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #17957
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/DataCollectorTranslatorPass.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->has('translator')) {
+            return;
+        }
+
+        $translatorClass = $container->findDefinition('translator')->getClass();
+
+        if (!is_subclass_of($translatorClass, '\Symfony\Component\Translation\TranslatorBagInterface')) {
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/DataCollectorTranslatorPass.php]: The leading `\` can be removed I think.

Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/DataCollectorTranslatorPass.php]: done

**Comment by fabpot** [comment]
Thank you @xabbuh.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/DataCollectorTranslatorPass.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ */
+class DataCollectorTranslatorPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->has('translator')) {
+            return;
+        }
+
+        $translatorClass = $container->findDefinition('translator')->getClass();
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/DataCollectorTranslatorPass.php]: you need to resolve the value, as the class may use a parameter

Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/DataCollectorTranslatorPass.php]: see #21680

**Pull request merged**